django/utils/http.py

```diff
@@ -232,30 +232,31 @@ def urlsafe_base64_decode(s):
 
 def parse_etags(etag_str):
     """
-    Parses a string with one or several etags passed in If-None-Match and
-    If-Match headers by the rules in RFC 2616. Returns a list of etags
-    without surrounding double quotes (") and unescaped from \<CHAR>.
+    Parses a string of ETags given in an If-None-Match or If-Match header as
+    defined by RFC 7232. Returns a list of quoted ETags, or ['*'] if all ETags
+    should be matched.
     """
-    etags = ETAG_MATCH.findall(etag_str)
-    if not etags:
-        # etag_str has wrong format, treat it as an opaque string then
-        return [etag_str]
-    etags = [e.encode('ascii').decode('unicode_escape') for e in etags]
-    return etags
+    if etag_str.strip() == '*':
+        return ['*']
+    else:
+        # This is just enough validation to prevent outdated RFC 2616 ETags
+        # from breaking things.
```

Comment:
Perhaps "breaking things" could be better described. Is the idea that they should still be matched?